src/common/containers/with-repository.js

```diff
@@ -22,7 +23,11 @@ function withRepository(WrappedComponent) {
     render() {
       const { fullName, ...passThroughProps } = this.props; // eslint-disable-line no-unused-vars
 
-      return (this.props.repository
+      // only render if current repo in the store matches repo in URL params
+      const { repository } = this.props;
+      const isReady = repository && repository.fullName === fullName;
+
+      return (isReady
```

Comment:
This data is not being fetched. It's built from URL params and set to the store.

So the owner/name pair from the URL is turned in the reducer into full repository object (with owner, name, fullName, url). 

This way WrappedComponent can rely on full repository object being available (and not parse it from URL itself).